demos/browser/app/meetingV2/component/ContentShareManager.ts

```diff
@@ -0,0 +1,271 @@
+        this.streamProvider = new FileMediaStreamProvider(ContentShareManager.TestVideo);
+        break;
+      }
+      case 'dropdown-item-content-share-test-mono-audio-speech': {
+        this.streamProvider = new AudioBufferMediaStreamProvider('audio_file', true);
+        break;
+      }
+      case 'dropdown-item-content-share-test-stereo-audio-speech': {
+        this.streamProvider = new AudioBufferMediaStreamProvider('stereo_audio_file', true);
+        break;
+      }
+      case 'dropdown-item-content-share-test-stereo-audio-tone': {
+        this.streamProvider = new SynthesizedStereoMediaStreamProvider(500, 1000);
+        break;
+      }
+      case 'dropdown-item-content-share-file-item': {
+        const fileList = document.getElementById('content-share-item') as HTMLInputElement;
+        const file = fileList.files[0];
+        if (!file) {
+          this.logger.error('No content share file selected');
+          return;
+        }
+        const url = URL.createObjectURL(file);
+        this.streamProvider = new FileMediaStreamProvider(url);
+        break;
+      }
+    }
+
+    this.transformedAudioStreamProvider = undefined;
+    this.transformedVideoStreamProvider = undefined;
+
+    if (this.enableCirculeCut && (await this.streamProvider.getMediaStream()).getVideoTracks().length !== 0) {
+      const stages = [new CircularCut()];
+      const videoTransformDevice = new DefaultVideoTransformDevice(
+        this.logger,
+        await this.streamProvider.getMediaStream(),
+        stages
+      )
+      this.transformedVideoStreamProvider = new VideoTransformDeviceMediaStreamProvider(this.streamProvider, videoTransformDevice);
+    }
+
+    if (this.enableVolumeReduction && (await this.streamProvider.getMediaStream()).getAudioTracks().length !== 0) {
+      this.transformedAudioStreamProvider = new AudioGainMediaStreamProvider(this.streamProvider, 0.1)
+    }
+
+    // if (false) {
+    if (this.transformedAudioStreamProvider || this.transformedVideoStreamProvider) {
+      this.streamProvider = new MergedMediaStreamProvider(
+        this.transformedAudioStreamProvider !== undefined ? this.transformedAudioStreamProvider : this.streamProvider, 
+        this.transformedVideoStreamProvider !== undefined ? this.transformedVideoStreamProvider : this.streamProvider)
+    }
+
+
+    if (this.streamProvider !== undefined) {
+      await this.audioVideo.startContentShare(await this.streamProvider.getMediaStream());
+    }
+
+    this.started = true;
+    this.paused = false;
+    this.updateContentShareUX();
+  }
+
+  async stop(): Promise<void> {
+    this.audioVideo.stopContentShare();
+    this.streamProvider?.pause();
+
+    this.started = false;
+    this.updateContentShareUX();
+  }
+
+  private initContentShareUI(): void {
+    const buttonContentShare = document.getElementById('button-content-share');
+    buttonContentShare.addEventListener('click', _e => {
+      AsyncScheduler.nextTick(() => {
```

Comment:
nit: I don't think we need to perform some operations in the next queue.

Reply:
Darn but I love unnecessarily asynchronous behavior. jk was from copy paste will remove.